kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java

```diff
@@ -5932,7 +5940,19 @@ public void commitResource(ContentResourceEdit edit, int priority) throws OverQu
                 //This might not want to be set as it would advise the detector
                 metadata.set(Metadata.RESOURCE_NAME_KEY, edit.getId());
                 metadata.set(Metadata.CONTENT_TYPE, currentContentType);
-                String newmatch = DETECTOR.detect(TikaInputStream.get(buff), metadata).toString();
+                String newmatch = "";
+                //If we are ignoring the content for this extension, don't give it any data
+                if (m_ignoreExtensions != null && m_ignoreExtensions.contains(FilenameUtils.getExtension(edit.getId()))) {
+                    newmatch = DETECTOR.detect(null, metadata).toString();
+                }
+                else {
+                  newmatch = DETECTOR.detect(TikaInputStream.get(buff), metadata).toString();
+                  //Redetect without the content as we're ignoring this mime type
+                  if (m_ignoreMimeTypes != null && m_ignoreMimeTypes.contains(FilenameUtils.getExtension(edit.getId()))) {
```

Comment:
You are looking for "FilenameUtils.getExtension(edit.getId())" in both lists "ignorecontent.extensions" and "ignorecontent.mimetypes", but one is a list of extensions and the other one is a list of mime types, I think this condition will never be true... Maybe this line should be:
```
if (m_ignoreMimeTypes != null && m_ignoreMimeTypes.contains(newmatch)) {
```